test/TEST-02-CRYPTSETUP/test.sh

```diff
@@ -1,38 +1,31 @@
 #!/bin/bash
 # -*- mode: shell-script; indent-tabs-mode: nil; sh-basic-offset: 4; -*-
 # ex: ts=8 sw=4 sts=4 et filetype=sh
+set -e
 TEST_DESCRIPTION="cryptsetup systemd setup"
+TEST_NO_NSPAWN=1
 
 . $TEST_BASE_DIR/test-functions
 
 check_result_qemu() {
     ret=1
     mkdir -p $TESTDIR/root
     mount ${LOOPDEV}p1 $TESTDIR/root
-    [[ -e $TESTDIR/root/testok ]] && ret=0
-    [[ -f $TESTDIR/root/failed ]] && cp -a $TESTDIR/root/failed $TESTDIR
+    [ ! -e $TESTDIR/root/testok ] || ret=0
```

Comment:
In https://tiswww.case.edu/php/chet/bash/bashref.html#The-Set-Builtin I've just read
> The shell does not exit if the command that fails is part of any command executed in a && or || list except the command following the final && or ||

and I think that `[! cmd1 ] || cmd2` should not be used instead of `[[ cmd1 ]] && cmd2`, because the return code of `cmd1` doesn't really matter, but `[! cmd1 ] || cmd2` makes it harder to read the code.